app/views/landing-pages/league/components/Leaderboard.vue

```diff
@@ -87,20 +102,23 @@ export default {
           th(colspan=1)
           th(colspan=1) {{ $t('general.rank') }}
           th {{ $t('general.score') }}
-          th.name-col-cell {{ $t('general.name') }}
+          th.name-col-cell(v-if="showStudentNames") {{ $t('general.username') }}
+          th.name-col-cell(v-else) {{ $t('general.name') }}
+          th.name-col-cell(v-if="showStudentNames") {{ $t('teacher.student_name') }}
           th(colspan=4 style="text-transform: capitalize;") {{ $t('league.team') }}
           th(colspan=1) {{ $t('ladder.age') }}
           th(colspan=1) 🏴‍☠️
 
       tbody
-        tr(v-for="row, rank in rankings" :key="rank" :class="isMyRow(row) ? 'success' : ''")
+        tr(v-for="row, rank in rankings" :key="rank" :class="classForRow(row)")
           template(v-if="row.type==='BLANK_ROW'")
             td(colspan=3) ...
           template(v-else)
             td.code-language-cell(:style="`background-image: url(/images/common/code_languages/${row.submittedCodeLanguage}_icon.png)`" :title="row.submittedCodeLanguage")
             td.rank-cell {{ row.rank || rank + 1 }}
             td.score-cell {{ scoreForDisplay(row) }}
             td(:class="'name-col-cell' + ((new RegExp('(Bronze|Silver|Gold|Platinum|Diamond) AI')).test(row.creatorName) ? ' ai' : '')") {{ row.creatorName || $t("play.anonymous") }}
+            td.name-col-cell(v-if="showStudentNames") {{ row.fullName || $t("teacher.not_applicable") }}
```

Comment:
I think we should be showing either the student full name or the username, not both. (And if we don't have a full name for some reason, we should fall back to the username.